### ISSUE TITLE
Closing the last open project closes the Dynamo Session

### ISSUE THREAD

**Issue by JacobSmall** [DynamoDS/DynamoRevit#2715]
## Dynamo version

2.6

## Revit version

2021

## Operating system

Windows

## What did you do? 

Close the last open Revit window to start a new project to test a graph on

## What did you expect to see?

Dynamo to still be open

## What did you see instead?

Dynamo closed. This leads to additional time spent relaunching Dynamo, prevent a user from opening an RVT, running a dyn, closing the rvt, opening a new rvt, running the dyn, closing the rvt, etc. 

Dynamo will hot swap between projects, so a better behavior is possible. to reproduce hot swapping:
1) Create a new project in Revit (project1) and launch Dynamo.
2) In Dynamo build a graph that does something simple (sandbox only - I did not test the Revit interaction).
3) Back in Revit make a new project file. Dynamo status: `Dynamo is not pointing at the current document.`
4) Switch back to the first Revit project. Dynamo status: `Dynamo is now available.`
5) Close the first Revit project. Dynamo Status: `Dynamo is not pointing at the current document.` flashes briefly before switchign back to `Dynamo is now available.

"As a Revit users I should not have to relaunch a tool each time I close my last project."

**Comment by QilongTang** [comment]
@ZiyunShang Do you recall the exact change/PR for this? From my memory, we force closing Dynamo because previously there was certain bugs associated.

**Comment by ZiyunShang** [comment]
Hi @QilongTang , I think you mean this PR - https://github.com/DynamoDS/DynamoRevit/pull/2317 . When Dynamo lost the host revit document, there will be a crash.

**Comment by QilongTang** [comment]
> Hi @QilongTang , I think you mean this PR - #2317 . When Dynamo lost the host revit document, there will be a crash.

Right, when Dynamo lost the host Revit document, and user press undo/redo, there will be a crash..
